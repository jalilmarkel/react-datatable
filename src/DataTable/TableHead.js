import styled from 'styled-components';

const TableHead = styled.div`
  display: flex;
  text-align: left;
  background-color: ${props => props.theme.header.backgroundColor};
`;

export default TableHead;
