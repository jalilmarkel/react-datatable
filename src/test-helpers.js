import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import theme from './themes/default';

export const renderWithTheme = (tree, ...args) =>
  render(<ThemeProvider theme={theme()}>{tree}</ThemeProvider>, ...args);

