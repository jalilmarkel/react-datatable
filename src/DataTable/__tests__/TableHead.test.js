import 'jest-styled-components';
import React from 'react';
import { cleanup } from 'react-testing-library';
import { renderWithTheme } from '../../test-helpers';
import TableHead from '../TableHead';

afterEach(cleanup);

test('component <TableHead /> should render correctly', () => {
  const { container } = renderWithTheme(<TableHead />);

  expect(container.firstChild).toMatchSnapshot();
});
