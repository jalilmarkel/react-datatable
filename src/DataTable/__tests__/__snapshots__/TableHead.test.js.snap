// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <TableHead /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: left;
  background-color: transparent;
}

<div
  class="c0"
/>
`;
